romancal/skycell/match.py

```diff
@@ -0,0 +1,225 @@
+        return self._radec_corners
+
+    @cached_property
+    def radec_center(self) -> tuple[float, float]:
+        """center point in right ascension and declination"""
+        return sgv.vector_to_lonlat(self.vectorpoint_center)
+
+    @cached_property
+    def vectorpoint_corners(self) -> NDArray[float]:
+        """corners in 3D Cartesian space on the unit sphere"""
+        return sgv.normalize_vector(
+            np.stack(sgv.lonlat_to_vector(*np.array(self.radec_corners).T), axis=1)
+        )
+
+    @cached_property
+    def vectorpoint_center(self) -> tuple[float, float, float]:
+        """center in 3D Cartesian space on the unit sphere"""
+        return sgv.normalize_vector(np.mean(self.vectorpoint_corners, axis=0))
+
+    @cached_property
+    def length(self) -> float:
+        """diagonal length of the rectangular footprint"""
+        # assume radial against sky background
+        return sga.length(self.vectorpoint_corners[0], self.vectorpoint_corners[2])
+
+    @cached_property
+    def circumference(self) -> float:
+        """circumference of the rectangular footprint"""
+        return sum(
+            sga.length(
+                self.vectorpoint_corners[index], self.vectorpoint_corners[index + 1]
+            )
+            for index in range(-1, len(self.vectorpoint_corners) - 1)
+        )
+
+    @cached_property
+    def polygon(self) -> sgp.SingleSphericalPolygon:
+        """spherical polygon representing this image footprint"""
+        return sgp.SingleSphericalPolygon(
+            points=self.vectorpoint_corners,
+            inside=self.vectorpoint_center,
+        )
+
+    @cached_property
+    def area(self) -> float:
+        """area of this footprint on the sphere in degrees squared"""
+        return self.polygon.area()
+
+    @cached_property
+    def possibly_intersecting_projregions(self) -> int:
+        """number of possibly intersecting projection regions"""
+        if self.area > sc.SkyCell.area:
+            return (
+                # the number of times the smallest projection region could fit in the image footprint
+                np.ceil(self.area / sc.ProjectionRegion.MIN_AREA)
+                # plus multiplier for partial intersections on the perimeter
+                * 8
+            )
+        else:
+            # 4 foundational intersections
+            return 4
+
+    @cached_property
+    def possibly_intersecting_skycells(self) -> int:
+        """number of possibly intersecting skycells"""
+        if self.polygon.area() > sc.SkyCell.area:
+            return (
+                # number of times a skycell could fit in the image footprint
+                np.ceil(self.polygon.area() / sc.SkyCell.area)
+                # plus multiplier for partial intersections on the perimeter
+                * 8
+            )
+        else:
+            # 4 foundational intersections
+            return 4
+
+
+def find_skycell_matches(
```

Comment:
Is there a use case for this? I was under the impression that all data images will be resampled onto sky cells. This code suggests that arbitrary mosaics may be so resampled, but I though that had been ruled out.

Reply:
We're only planning on resampling images onto skycells, and not arbitrary mosaics, so I agree that this code is doing more than is needed, though that may be fine for now.